river/command/rule.zig

```diff
@@ -95,6 +100,37 @@ pub fn ruleAdd(_: *Seat, args: []const [:0]const u8, _: *?[]const u8) Error!void
                 .value = output_name,
             });
         },
+        .position => {
+            const x = try fmt.parseInt(c_int, result.args[1], 10);
+            const y = try fmt.parseInt(c_int, result.args[2], 10);
+            try server.config.position_rules.add(.{
+                .app_id_glob = app_id_glob,
+                .title_glob = title_glob,
+                .value = .{
+                    .x = x,
+                    .y = y,
+                },
+            });
+        },
+        .dimensions => {
+            const width = try fmt.parseInt(c_int, result.args[1], 10);
+            const height = try fmt.parseInt(c_int, result.args[2], 10);
+            try server.config.dimensions_rules.add(.{
```

Comment:
I think we need to error on negative x/y and definitely on negative width/height. A simple way to do this is to pass the type `u31` to `parseInt()` instead of `c_int`.

Reply:
I opted to check if x/y and width/height are negative explicitly in order to assign a more appropriate error `Error.InvalidValue`. I also added the non-negative requirement to the riverctl documentation.

Reply:
I don't think I agree, "out of bounds" is a bit more descriptive than "invalid value" IMO and is what we use elsewhere in river for exactly this situation.

Reply:
I see, I should have looked for precedence in the source. I have updated the block to pass the type `u31` to `parseInt()` as well as the field types in the `Position` and `Dimensions` struct.